### DISCUSSION TITLE
Direction of React Router v7

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#5774] [Q&A]
Hi there,

Recently in the React conference, Remix annouced the v7 roadmap which is basically React Router + Vite = v7. I wonder how this will have impact on Modern js or any Rspack based project.

thanks

**Comment by yimingjfe** [comment]
Thank you for your feedback, we also noticed this change in react-router.

For Modern.js users, I don't think there will be any impact, we will approach this change cautiously to make sure it wouldn't be troublesome for current users, and I've also seen the Remix team promise that v6 to v7 will be a [non-breaking upgrade](https://remix.run/blog/incremental-path-to-react-19#tldr).

Regarding the framework evolution, we don't know some of the details of react-router v7 yet, so I can't give you exactly some information, but in general we want to continue using react-router and work intensively with the Remix team to bring some of the great features of react-router v7 to Modern.js users.
